SWLOR.Game.Server/Native/GetDamageRoll.cs

```diff
@@ -281,50 +310,61 @@ private static int OnGetDamageRoll(void* thisPtr, void* pTarget, int bOffHand, i
                 }
-                else if (damageType == CombatDamageType.Force && damage > 0)
-                {
-                    ApplyEffectToObject(DurationType.Instant, EffectDamage(damage, DamageType.Divine), targetObject.m_idSelf);
-                    DoFeedback(attacker.m_idSelf, targetObject.m_idSelf, "Force", damage, 255,255,255);
-                }
-                else if (damageType == CombatDamageType.Fire && damage > 0)
-                {
-                    ApplyEffectToObject(DurationType.Instant, EffectDamage(damage, DamageType.Fire), targetObject.m_idSelf);
-                    DoFeedback(attacker.m_idSelf, targetObject.m_idSelf, "Fire", damage, 255,0,0);
-                }
-                else if (damageType == CombatDamageType.Poison && damage > 0)
-                {
-                    ApplyEffectToObject(DurationType.Instant, EffectDamage(damage, DamageType.Acid), targetObject.m_idSelf);
-                    DoFeedback(attacker.m_idSelf, targetObject.m_idSelf, "Poison", damage, 0,255,0);
-                }
-                else if (damageType == CombatDamageType.Electrical && damage > 0)
-                {
-                    ApplyEffectToObject(DurationType.Instant, EffectDamage(damage, DamageType.Electrical), targetObject.m_idSelf);
-                    DoFeedback(attacker.m_idSelf, targetObject.m_idSelf, "Electrical", damage, 0,0,255);
-                }
-                else if (damageType == CombatDamageType.Ice && damage > 0)
+                else
                 {
-                    ApplyEffectToObject(DurationType.Instant, EffectDamage(damage, DamageType.Cold), targetObject.m_idSelf);
-                    DoFeedback(attacker.m_idSelf, targetObject.m_idSelf, "Ice", damage, 0,255,255);
+                    // Assign elemental damage.  We want to do so on a delay because otherwise the damage (and possible target death)
+                    // show up in the combat log before the attack roll (!).  Unfortunately, DelayCommand and AssignCommand->DelayCommand
+                    // don't work here. TODO: find out how to assign/delay commands in native. 
+                    // Alternately, we could find out how to access the damage in a structure like NWNX_Damage and just enter the 
+                    // damage types there directly. 
+                    ProcessElementalDamage(damageType, damage, attacker, targetObject);
                 }
-
             }
 
             return physicalDamage;
         }
 
+        private static void ProcessElementalDamage(CombatDamageType damageType, int damage, CNWSCreature attacker, CNWSObject targetObject)
+        {
+            if (damageType == CombatDamageType.Force && damage > 0)
+            {
+                DoFeedback(attacker.m_idSelf, targetObject.m_idSelf, "Force", damage, 255, 255, 255);
+                ApplyEffectToObject(DurationType.Instant, EffectDamage(damage, DamageType.Divine), targetObject.m_idSelf);
```

Comment:
Is it safe to make this call? We're in a native context so I'm worried we don't have direct access to NWScript at this point.

Reply:
I replaced this in the 'combine all damage types into a single number' commit.  It's safe to call (as in it works and doesn't crash anything) but dealing damage from here has some side effects around timing that make it awkward.